modules/com_vtiger_workflow/VTSimpleTemplate.inc

```diff
@@ -593,6 +596,36 @@ class VTSimpleTemplate {
 				} else {
 					return '';
 				}
+			case 'scanBarCode':
+				$wsId = $this->parent->getId();
+				$parts = explode('x', $wsId);
+				$recordId = $parts[1];
+				$moduleName = $this->parent->getModuleName();
+				$focus = CRMEntity::getInstance($moduleName);
+				$focus->retrieve_entity_info($recordId, $moduleName);
+				if (!empty($focus->column_fields[$field])) {
+					$tcpdf1dEncodingTypes = array('EAN8' => 'EAN8','EAN13' => 'EAN13');
+					$tcpdf2dEncodingTypes = array('QRCODE,L' => 'QRCODE,L','QRCODE,M' => 'QRCODE,M','QRCODE,Q' => 'QRCODE,Q','QRCODE,H' => 'QRCODE,H');
+					$fieldValue = $focus->column_fields[$field];
+
+					require_once 'include/tcpdf/tcpdf_barcodes_2d.php';
+					require_once 'include/tcpdf/tcpdf_barcodes_1d.php';
+
+					if (array_key_exists($codetype, $tcpdf1dEncodingTypes)) {
+						$barcodeobj = new TCPDFBarcode($fieldValue, $codetype);
+					} elseif (array_key_exists($codetype, $tcpdf2dEncodingTypes)) {
+						$barcodeobj = new TCPDF2DBarcode($fieldValue, $codetype);
+					}
+					$img = $barcodeobj->getBarcodePngData();
+					$fname = uniqid('barcode');
+					@unlink('cache/images/'.$fname.'.png');
+					file_put_contents('cache/images/'.$fname.'.png', $img);
+					return '<img src="cid:'.$fname.'" />';
+					print_r('<img src="cid:'.$fname.'" />');
+					die();
```

Comment:
please eliminate these two lines and I will accept

Reply:
Done